pkg/controller/operators/catalog/operator.go

```diff
@@ -209,9 +210,43 @@ func NewOperator(kubeconfigPath string, logger *logrus.Logger, wakeupInterval ti
 		OpClient:             op.OpClient,
 		Lister:               op.lister,
 	}
+
+	// Namespace sync for resolving subscriptions
+	namespaceInformer := informers.NewSharedInformerFactory(op.OpClient.KubernetesInterface(), wakeupInterval).Core().V1().Namespaces()
+	resolvingNamespaceQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "resolver")
+	namespaceQueueInformer := queueinformer.NewInformer(
+		resolvingNamespaceQueue,
+		namespaceInformer.Informer(),
+		op.syncResolvingNamespace,
+		nil,
+		"resolver",
+		metrics.NewMetricsNil(),
+		logger,
+	)
+
+	op.RegisterQueueInformer(namespaceQueueInformer)
+	op.lister.CoreV1().RegisterNamespaceLister(namespaceInformer.Lister())
+	op.namespaceResolveQueue = resolvingNamespaceQueue
+
+	// Create an informer/lister for operatorgroups
+	for _, namespace := range watchedNamespaces {
+		sharedInformerFactory := externalversions.NewSharedInformerFactoryWithOptions(crClient, wakeupInterval, externalversions.WithNamespace(namespace))
+		operatorGroupInformer := sharedInformerFactory.Operators().V1alpha2().OperatorGroups()
+		op.lister.OperatorsV1alpha2().RegisterOperatorGroupLister(namespace, operatorGroupInformer.Lister())
+
+		// Register queue and QueueInformer
+		queueName := fmt.Sprintf("%s/operatorgroups", namespace)
+		operatorGroupQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), queueName)
+		operatorGroupQueueInformer := queueinformer.NewInformer(operatorGroupQueue, operatorGroupInformer.Informer(), NoopHandler, nil, queueName, metrics.NewMetricsNil(), logger)
+		op.RegisterQueueInformer(operatorGroupQueueInformer)
+	}
 	return op, nil
 }
 
+func NoopHandler(obj interface{}) error {
```

Comment:
Could just use an anonymous function instead since this is only used once.

Reply:
I kept this around in the git history but I removed the gate on operatorgroup. It didn't end up being related to the fix in this PR (added it to verify it wasn't an issue originally), and I don't think it provides the best UX for that scenario (an error message in a log)